File: InterviewPreparationKits/PlusMinus.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'plusMinus' function below.
#
# The function accepts INTEGER_ARRAY arr as parameter.
#

def plusMinus(arr):
    positives = 0
    zeros = 0
    negatives = 0
    for i in range(len(arr)):
        arr[i] = round(arr[i]/len(arr),6)
        if arr[i] > 0: positives += 1
        elif arr[i] == 0:
            zeros += 1
        else: negatives+=1
    print(positives/len(arr))
    print(negatives/len(arr))
    print(zeros/len(arr))
    
    

if __name__ == '__main__':
    n = int(input().strip())

    arr = list(map(int, input().rstrip().split()))

    plusMinus(arr)

File: InterviewPreparationKits/PlusMinus.js
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', function(inputStdin) {
    inputString += inputStdin;
});

process.stdin.on('end', function() {
    inputString = inputString.split('\n');

    main();
});

function readLine() {
    return inputString[currentLine++];
}

/*
 * Complete the 'plusMinus' function below.
 *
 * The function accepts INTEGER_ARRAY arr as parameter.
 */

function plusMinus(arr) {
    let positives = 0;
    let negatives = 0;
    let zero = 0;
    for(let i = 0; i<arr.length; i++){
        if (arr[i] == 0)
            zero++;
        if(arr[i] > 0)
            positives++;
        if(arr[i] < 0)
            negatives++;            
    }
    let num = arr.length;
    console.log((positives/num).toFixed(6));
    console.log((negatives/num).toFixed(6));
    console.log((zero/num).toFixed(6));

}

function main() {
    const n = parseInt(readLine().trim(), 10);

    const arr = readLine().replace(/\s+$/g, '').split(' ').map(arrTemp => parseInt(arrTemp, 10));

    plusMinus(arr);
}
